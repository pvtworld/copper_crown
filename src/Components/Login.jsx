import React from 'react';
let {Link, IndexLink} = require('react-router');



class Login extends React.Component{
    render() {
        return (
            <div>
                <h2>Login Component</h2>
                <div>
                    <Link to="/coppermap" activeClassName="active"  activeStyle={{fontWeight: 'bold'}}>CopperMap</Link>
                </div>
                <div>
                    <button type="button" className="btn btn-default navbar-btn">Sign in with Github</button>
                </div>
            </div>
        );
    }
}

export default Login;


