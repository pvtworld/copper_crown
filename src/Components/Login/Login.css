.login{
    width: 400px;
    margin: 0 auto;
}

p {
    color: white;
}

.box{
    margin: 80px auto 10px;
    width:400px;
    padding: 20px 20px 20px;
    background: rgba(0,0,0,0.6);
    position: relative;
    border-radius: 5px;
    -webkit-border-radius: 5px;
    -moz-border-radius: 5px;
}

@media screen and (max-width: 540px) {
    .login{
        width: 100%;
    }
}



