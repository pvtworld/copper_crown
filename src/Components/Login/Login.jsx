import React from 'react';
import {compose} from 'redux';
import { connect } from 'react-redux';
import { firebaseConnect } from 'react-redux-firebase'
import './Login.css'
import {Button, Panel, Col} from 'react-bootstrap';
import ToolTip from '../ToolTip/ToolTip'
import { loadingUser, finishedLoadingUser, loadingError } from '../../Redux/Actions/copperMapActions'
import { browserHistory } from 'react-router';

const title = <h2>Sign in to play the game</h2>;

const loginUser = (firebase, provider, dispatch) => {
    dispatch(loadingUser());
    firebase.login({provider: provider, type: 'popup'}).then(() => {
        dispatch(finishedLoadingUser());
        console.log('Login successful, redirecting to /app');
        browserHistory.push('/app')
    }, (error) => {
        //handle login error
        dispatch(loadingError());
        alert('Unable to authenticate!\n' + error);
    })
};

const Login = (props) => {
        return (
            <Col md={4} mdOffset={4}>
                <Panel className="login" header={title}>
                    <Button bsStyle="primary" bsSize="large" block onClick={() => loginUser(props.firebase, 'facebook', props.dispatch)} >Log In with Facebook</Button>
                    <Button bsStyle="default" bsSize="large" block onClick={() => loginUser(props.firebase, 'github', props.dispatch)}>Log In with Github</Button>
                    <Button bsStyle="danger" bsSize="large" block onClick={() => loginUser(props.firebase, 'google', props.dispatch)} >Log In with Google</Button>
                    <Col className="text-right">
                        <ToolTip delayShow={200} id="tooltip-bottom" placement="bottom" tooltip="Help text. Help text. Help text. Help text. Help text. Help text." >Help</ToolTip>
                    </Col>
                </Panel>
            </Col>
        );
    };

export default compose(firebaseConnect(), connect())(Login)


