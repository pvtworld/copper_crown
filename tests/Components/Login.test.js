import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Login from '../../src/Components/Login/Login';

beforeAll(() => {
    
});

describe('Render test', ()=> {
    it('Renders without crashing', () => {
        const component = renderer.create(<Login />);
        
        expect(component).toMatchSnapshot();
    });
});