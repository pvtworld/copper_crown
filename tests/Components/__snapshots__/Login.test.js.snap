exports[`Render test Renders without crashing 1`] = `
<div
  className="custom"
  style={
    Object {
      "margin": "40px auto 10px",
      "maxWidth": 400,
    }
  }>
  <h4>
    Login with
  </h4>
  <button
    className="btn btn-lg btn-primary btn-block"
    disabled={false}
    onClick={[Function]}
    type="button">
    Facebook
  </button>
  <button
    className="btn btn-lg btn-default btn-block"
    disabled={false}
    onClick={[Function]}
    type="button">
    Github
  </button>
  <button
    className="btn btn-lg btn-danger btn-block"
    disabled={false}
    onClick={[Function]}
    type="button">
    Google
  </button>
</div>
`;
